deployer/scripts/install.sh

```diff
@@ -255,7 +255,13 @@ function generate_configmaps() {
     # generate curator configmap
     oc create configmap logging-curator \
       --from-file=config.yaml=conf/curator.yml
-    oc label configmap/logging-curator logging-infra=support # make easier to delete later
+    oc label configmap/logging-curator logging-infra=support
+
+    # generate fluentd configmap
+    oc create configmap logging-fluentd \
+      --from-file=fluent.conf=conf/fluent.conf \
+      --from-file=throttle-config.yaml=conf/fluentd-throttle-config.yaml || return
```

Comment:
In this PR, the throttle configs are dynamically generated from a configmap.

There is a delicate balance here in how much we allow users to edit; after
all, we're responsible for providing sane configuration. Users want to edit
everything, but we can't really support them editing *anything* if we want
to hold open the option of updates in the future.

The middle ground I've proposed here is to provide most of the config on
the image (the configs.d/openshift directory which we'll own updates for)
but provide a configmap for editing the fluent.conf that brings it all
together with @include (which we hope not to change substantially for the
future). You can change the provided files by adding your own version into
the configmap and editing fluent.conf to point at it. Doing so will provide
a clear separation; if we update our image and change something about the
file they "edited" then it will be easy to figure out what happened and fix
it.

I'm honestly a bit nervous about even making fluent.conf configurable, but
it has already proved really useful in my testing, and I know customers
will want *some* power to modify config without having to incorporate a
custom image.

On Wed, Jul 13, 2016 at 10:25 AM, Anton <notifications@github.com> wrote:

> In deployer/scripts/install.sh
> <https://github.com/openshift/origin-aggregated-logging/pull/195#discussion_r70635007>
> :
>
> > @@ -255,7 +255,13 @@ function generate_configmaps() {
> >      # generate curator configmap
> >      oc create configmap logging-curator \
> >        --from-file=config.yaml=conf/curator.yml
> > -    oc label configmap/logging-curator logging-infra=support # make easier to delete later
> > +    oc label configmap/logging-curator logging-infra=support
> > +
> > +    # generate fluentd configmap
> > +    oc create configmap logging-fluentd \
> > +      --from-file=fluent.conf=conf/fluent.conf \
> > +      --from-file=throttle-config.yaml=conf/fluentd-throttle-config.yaml || return
>
> Why not move hardcoded fluentd config files and related env variables to
> configmaps as well?
> Is there a plan to change dynamically generated throttled configs to
> configmaps?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin-aggregated-logging/pull/195/files/a46c725bc59c9ac2d21d547550c583493e5b9ce7#r70635007>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AABz-gPMpLJU48aPx6ksVMlboo-7Yx0Aks5qVPV2gaJpZM4JHWV8>
> .
>